### ISSUE TITLE
Comprehensive Suggestions for Exo Armor

### ISSUE THREAD

**Issue by Divineaspect** [Esteemed-Innovation/Flaxbeards-Steam-Power#337]
I put quite a bit of thought into this, and gathered them all into one place as requested. 
The focus of this is on expanding existing concepts already implemented, with the goal of three good component choices which aren't cross mod.

**Armor Plates**
* [x] Gilded Iron Plate: Weaker combo of Iron and Gold plates.
  * [x] Code
  * [x] Texture
  * [x] Model
  * [x] Recipe
  * [x] Lang
* [x] Zinc Armor Plate: Ablative! Pops off when I would get below 5 hearts, and prevents up to 5 hearts of damage.
  * [x] Code
  * [x] Texture
  * [x] Model
  * [x] Recipe
  * [x] Lang

**Helmet Slot**
* Conical Hat: Deflects falling items, and breaks them if they would become solid over your head, Great for people who keep digging up.
* Finned Helmet (I just watched the rocketeer): makes me go faster? or jerk in the direction I suddenly twitch the mouse?
* [x] Frequency Shifter: This odd acoustic invention Changes the frequencies you are hearing and speaking on. As a result you can trade with cats and dogs in their preferred currencies, and you don’t hear chat messages except from others wearing the shifter.
  * [x] Code
  * [x] Item Texture
  * [x] Model
  * [x] Recipe
  * [x] Lang

**Mask Slot**
* [x] Rebreather: Restores breath bubbles at a steam cost.
  * [x] Code
  * [x] Item Texture
  * [x] Armor Texture
  * [x] Recipe
  * [x] Lang
* [x] Dragon’s roar: this Emergency Steam Vent outward over the mouth provides the ability to damage a whole group of enemies.
  * [x] Code
  * [x] Item Texture
  * [x] Model
    * Copper robotic dragon head mask
  * [x] Recipe
  * [x] Lang

**Arm Slot**
* [x] Piston Push: While not as satisfying as the powerfist, this simple brass piston allows me to move blocks away from me as if I was a piston. Surprisingly useful in exploring for caves while mining.
  * [x] Code
  * [x] Item Texture
  * [x] Armor Texture
  * [x] Recipe
  * [x] Lang
* Smasher Punch: Mounting a rock smasher on my arm was a brilliant idea, now I don’t need to spend as much time processing, as ores I break are already smashed into easily blendable components.

**Hip Slot**
* Recharger: Drains your Steam into steamholding items, greatly expanding the field time for steam powered tools.
* [x] Reloading Holsters: Reloads your pistols while you aren’t holding them.
  * [x] Code
  * [x] Item Texture
  * [x] Model
  * [x] Recipe
  * [x] Lang
* Secondary Steam Tanks: Adds basic steam storage to your pants!

**Leg Slot**
* Gyroscopes: Keeping me steady, and removing nausea and 

**Heel Slot**
* [x] Anchor Heels: Lead Plates help hold me down, and rapidly sink me to the bottom of the ocean, as well as resist knockback.
  * [x] Code
  * [x] Item Texture
  * [x] Model
  * [x] Recipe
  * [x] Lang
* Snowshoes: Walk up snow and ice blocks as if they were stairs. Great for if you’ve built your lab on a mountaintop.

**FootSlot**
* [x] Hydrophobic Coatings: Zinc and Slime exert water repulsive properties when heated, this should allow me to coast (as if sliding on ice) over the surface of water.
  * [x] Code
  * [x] Item Texture
  * [x] Armor texture
  * [x] Recipe
  * [x] Lang
* [x] Pyrophobic Coatings: Copper and Magma Creme, as above, only on lava and a lot more expensive.
  * [x] Code
  * [x] Item Texture
  * [x] Armor texture
  * [x] Recipe
  * [x] Lang

**Comment by elifoster** [comment]
What does the Gilded Iron Plate do?
What could the Zinc Plate's lore be?
For the Gyroscope, was there another idea? There's an 'and', but no following ideas.
Isn't the Snowshoes already possible with just the exo feet?

For ideas I like, I will add a checkbox.

**Comment by Divineaspect** [comment]
Gilded Iron: I was thinking of combining iron and gold's properties, as a way of encouraging people to use the FSP Guns, but if that's too powerful I'll come up with something else.

Zinc Plate Lore: "From studying the function of the rupture disc, I've learned how to make an armor plate which contains steam within it, when sufficiently damaged it explodes off of my body more then negating the force of a potentially critical blow."

Gyroscope: My other original idea is what that it prevented people from falling off edges, as per crouching, but that was potentially aggravating.

Snowshoes: actually the stacking semiblocks of snow can mess it up.

Reloading Hoster: I have seriously tried to ONLY use FSP guns and go on an adventure... It was painful.

Secondary Steam Tanks: But what if I want to wear some other chestpiece?

**Comment by Flaxbeard** [comment]
I'm just chiming in but I think the reloading holsters are a hilarious concept to imagine  (and also functionally pretty cool)

**Comment by elifoster** [comment]
I like the Gilded Iron. Maybe it should be a combination of the two effects, but slightly weaker.

Gyroscope sounds more annoying than anything.

Snowshoes sounds like a bug needs to be fixed.

Reloading Holster sounds pretty good now that I'm thinking about it.

Secondary Steam Tanks I'm torn about.

**Comment by elifoster** [comment]
So the Zinc Plate is almost working. The only thing I'm struggling with is how to actually update the ItemStack when the NBTTag "plates" is removed. It successfully gets removed, but the Zinc Exosuit Plate item never actually gets removed. @Flaxbeard The code is super similar to the UtilEnhancement#removeEnhancement code, perhaps you could take a look to see if you see anything I'm doing wrongly? Anyway, we still need an item sprite for the Rebreather @Esteemed-Innovation/aesthetics-team

**Comment by elifoster** [comment]
@Divineaspect the only way I'm able to get the -Phobic Coatings to work is requiring the user to jump off of the water. Setting the step height doesn't affect anything, probably because they aren't technically walking on a block, so the only way I'm able to reasonably get the user out of the water is by setting their motionY to something greater than 0 when they press the jump key.

**Comment by Divineaspect** [comment]
have it make a phantom block the thickness of a pressureplate, which
evaporates as they leave.

Jeremiah Grimsley


"All of humanity is in peril of extinction if each one of us does not dare,
now and henceforth, always to tell only the truth, and all the truth, and
to do so promptly — right now." - R Buckminster Fuller

**Comment by elifoster** [comment]
I did that, and it was really jumpy and overly complicated.

**Comment by elifoster** [comment]
Oh by the way, the Anchor Heels are crossmod

**Comment by Divineaspect** [comment]
Anchor heels alternate recipie, anvil...

**Comment by elifoster** [comment]
@Divineaspect Do you have ideas for all the recipes? I'm not very good with recipes. The only one I have so far is for the Anchor Heels:
* Standard lead recipe:
```
LeadPlate |       | LeadPlate
-------------------------------
LeadExo   |       | LeadExo
```

* Anvil recipe:
```
IronPlate |       | IronPlate
-----------------------------
IronExo   | Anvil | IronExo
```

**Comment by elifoster** [comment]
Just so everyone knows, I just rebased divine-exo-upgrades with master. This required a push -f, so you'll want to re-get the branch if you have it locally.

**Comment by elifoster** [comment]
So @Esteemed-Innovation/lead-developers, @Esteemed-Innovation/aesthetics-team, and @Esteemed-Innovation/testing-monkeys should we do the Conical Hat? There were some discussions on IRC that lead to a no, but I'm not sure.

**Comment by elifoster** [comment]
How should the Dragon Roar be triggered? Keybind? Automatically? Those are the only ones I can think of.

**Comment by Divineaspect** [comment]
Dragon Roar: Crouch and open hand click to represent clicking the valve?

**Comment by elifoster** [comment]
Well, all of the code is done for this issue. Now we're just waiting on the @Esteemed-Innovation/aesthetics-team for some textures and models.  Here are some more in-depth descriptions of the things:
* Rebreather: We already have the model, so all we need is the item texture. [this](https://raw.githubusercontent.com/Esteemed-Innovation/Flaxbeards-Steam-Power-Resources/8d2f716ca6c447f270a184e366b373ea9e355208/src/main/resources/assets/steamcraft/textures/models/armor/rebreatherUpgrade.png) in item form?
* Dragon Roar: Like an Ender Dragon head, but so that it fits over the head so you can still see mask upgrades. Coppery orange.
* Piston Push: Like the Power Fist, but with a pushing pistony thing. Needs some sort of animation to simulate pushing things forward.

As for the Frequency Shifter, Reloading Holsters, Anchor Heels, Pyrophobic Coatings, and Hydrophobic Coatings, I think we'll need to ask @Divineaspect

**Comment by Divineaspect** [comment]
Where are did the chosen recipes for those items I'm supposed to provide suggestions for the appearance of go?

**Comment by elifoster** [comment]
You mean [this](http://ftb.gamepedia.com/User:TheSatanicSanta/Sandbox/FSPRecipes)

**Comment by Divineaspect** [comment]
Frequency shifter: upside down headphones with the connecting band going under the chin, with a redstone antenna on one side.

Reloading Holsters: belt and holster with a brass buckle. The holster could have a revolving chamber, if it can be animated just make it spin when active, with a brass piston pointed up through the back of the chamber.

Anchor Heels: four heavy looking metal "toes", two in the front, two in the back of each foot.

Phobic coatings should appear to have the approprate metal plate texture on the sides and bottom, with streaks of slime and magma cream colors, add in a mini version of the heater on the outside. edges (facing downwards?)

**Comment by elifoster** [comment]
divine-exo-upgrades needs to be merged into master before the recipes for #251 can be made, as some require components added by this. @Lance5057 This is higher priority than #251.

**Comment by Divineaspect** [comment]
Model Thoughts
Frequency shifter: Earmuffs with an antenna like a redstone torch on one side
Anchor Heels: Just a set of "toes" at the foot area

**Comment by elifoster** [comment]
I have rebased again. All code is done except for models :smile:
